pybaseball/split_stats.py

```diff
@@ -0,0 +1,129 @@
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+from .datahelpers import postprocessing
+import bs4 as bs
+import pandas as pd
+import re
+
+
+def download_url(url: str) -> bytes:
+    """
+    Gets the content from the url specified
+    """
+    session = requests.Session()
+    retry = Retry(connect=3, backoff_factor=0.5)
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount('http://', adapter)
+    session.mount('https://', adapter)
+    resp = session.get(url)
+    return resp.content
+
+
+def get_split_soup(playerid: str, year: int = None, pitching_splits: bool = False) -> bs.BeautifulSoup:
+    """
+    gets soup for the player splits. 
+    """
+    pitch_or_bat = 'p' if pitching_splits else 'b'
+    if year is None:  # provides scores from yesterday if date is not provided
+        url = f"https://www.baseball-reference.com/players/split.fcgi?id={playerid}&year=Career&t={pitch_or_bat}"
+    else:
+        year = str(year)
+        url = f"https://www.baseball-reference.com/players/split.fcgi?id={playerid}&year={year}&t={pitch_or_bat}"
+    html = download_url(url)
+    soup = bs.BeautifulSoup(html, 'lxml')
+    return soup
+
+
+def get_player_info(playerid: str = None, soup=None) -> dict:
+    '''
+    Returns a dictionary with player position, batting and throwing handedness, player height in inches, player weight, and current team from Baseball Reference. 
+    '''
+
+    if not soup:
+        soup = get_split_soup(playerid)
+    about_info = soup.find_all(
+        "div", {"itemtype": "https://schema.org/Person"})
+    info = [ele for ele in about_info]
+    # This for loop goes through the player bio section at the top of the splits page to find all of the <p> tags
+    for i in range(len(info)):
+        ptags = info[i].find_all('p')
+        fv = []
+        # This loop goes through each of the <p> tags and finds all text between the tags including the <strong> tags.
+        for j in range(len(ptags)):
+            InfoRegex = re.compile(r'>(.*?)<', re.DOTALL)
+            r = InfoRegex.findall(str(ptags[j]))
+            # This loop cleans up the text found in the outer loop and removes non alphanumeric characters.
+            for k in range(len(r)):
+                pattern = re.compile('[\W_]+')
+                strings = pattern.sub(' ', r[k])
+                if strings and strings != ' ':
+                    fv.append(strings)
+    player_info_data = {
+        'Position': fv[1],
+        'Bats': fv[3],
+        'Throws': fv[5],
+        'Height': int(fv[6].split(' ')[0])*12+int(fv[6].split(' ')[1]),
+        'Weight': int(fv[7][0:3]),
+        'Team': fv[10]
+    }
+    return player_info_data
+
+
+def get_splits(playerid: str, year: int = None, player_info: bool = False, pitching_splits: bool = False) -> pd.DataFrame:
```

Comment:
To do this make sure you have `from typing import Optional` at the top

```suggestion
def get_splits(playerid: str, year: Optional[int] = None, player_info: bool = False, pitching_splits: bool = False) -> pd.DataFrame:
```